test/CMakeLists.txt

```diff
@@ -37,7 +37,8 @@ target_compile_definitions(test-sfml-system PRIVATE
     EXPECTED_SFML_VERSION_MAJOR=${SFML_VERSION_MAJOR}
     EXPECTED_SFML_VERSION_MINOR=${SFML_VERSION_MINOR}
     EXPECTED_SFML_VERSION_PATCH=${SFML_VERSION_PATCH}
-    EXPECTED_SFML_VERSION_IS_RELEASE=$<IF:$<BOOL:${VERSION_IS_RELEASE}>,true,false>
+    # Intentionally not using VERSION_IS_RELEASE_bool
+    "EXPECTED_SFML_VERSION_IS_RELEASE=\$<IF:\$<BOOL:${VERSION_IS_RELEASE}>,true,false>"
```

Comment:
Why are the quotes needed?

Reply:
Oh, it's a general thing I do. If you always quote arguments with variable substitution, except when you explicitly expect/need list expansion, then it's easy to spot those cases.

Bash works in the same way. I think some code is better to illustrate this:

```bash
A=""
B="a b"

num () { printf %d\\n "$#"; }

set -x

num $A # prints 0
num $B # prints 2

num "$A" # prints 1
num "$B" # prints 1
```

As you can see, in the first set of calls list expansion occurs, while in the second calls the argument is always received as exactly one argument.

Reply:
So you're saying we could remove those quotes and backslashes and this would still work? I think it's safe to assume that `VERSION_IS_RELEASE` shouldn't expand into a list and if it does, I'm fine with configuration failing.

Reply:
Being consistent with quoting arguments containing variable substitution expresses intent better (wanting list expansion vs not). Quoting has no downsides and avoids accidental issues with list expansion in places where it was not intended.

This is not purely a stylistic choice, so I want to make sure this is acknowledged.

Reply:
> expresses intent better (wanting list expansion vs not)

It expresses to me that you _need_ list expansion or not. That's why I first question was to ask why the quotes were needed thinking surely it was a technical necessity otherwise you would have omitted them.

> Quoting has no downsides

The downside is the code being a bit more complicated than necessary since parts of the generator expression must be escaped.

Reply:
Escaping the dollar is not mandatory, but I like to do it. Guess I'll undo that part.